modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMediator.java

```diff
@@ -410,7 +410,7 @@ private void processJSONPayload(MessageContext synCtx, ScriptMessageContext scri
                     jsonObject = this.jsEngine.eval('(' + scriptWithJsonParser + ')');
                 }
             } catch (ScriptException e) {
-                throw new SynapseException("Invalid JSON payload", e);
+                throw new ScriptException("Invalid JSON payload");
```

Comment:
By not showing the `e`, is there any useful information get hidden, if so better to keep the e

Reply:
Also, would it be helpful if we log the payload?

Reply:
IMO, logging the payload is not useful.

Reply:
With the e it propagates only the line number and column number where the error occurs so in this case the important thing to propagate would be that the payload is a invalid one.